src/Context/ThemeContext.tsx

```diff
@@ -0,0 +1,33 @@
+
+import { createContext, useState , ReactNode} from 'react';
+
+export interface ITheme{
+    theme: string;
+    setTheme: React.Dispatch<React.SetStateAction<string>>;
+    toggleTheme: () => void;
+}
+interface ThemeContextProviderProps {
+    children: ReactNode;
+  }
+export const ThemeContext = createContext<ITheme | null>(null);
+
+const ThemeContextProvider: React.FC<ThemeContextProviderProps>= (props)=>{
+    const [theme, setTheme] = useState('light');
+
+   const toggleTheme = () =>{
+        setTheme((current)=> (current === 'light' ? 'dark' : 'light' ))
```

Comment:
This is a valid approach, but it is more common to make the state variable a boolean value [isDarkMode, setIsDarkMode]
In this case, the toggling will be much simpler
`setTheme((previousValue)=> !previousValue)`